homework1/4var/main.py

```diff
@@ -0,0 +1,97 @@
+# Вариант 4
+# Дана целочисленная квадратная матрица. Определить:
+# 1) произведение элементов в тех строках, которые не содержат отрицательных элементов;
+# 2) максимум среди сумм элементов диагоналей, параллельных главной диагонали матрицы.
+import random
+
+
+def create_random_matrix(num_rows, num_columns):
+    matrix = [[0 for _ in range(num_columns)] for _ in range(num_rows)]
+    for i in range(num_rows):
+        for j in range(num_columns):
+            # matrix[i][j] = random.choice([-1, 1]) * random.randint(1, 100)
+            matrix[i][j] = random.randint(-50, 200)
+    return matrix
```

Comment:
it looks like you create list of zeros and then you assign some random values.
skip step with zero filing, just fill the list with random values